core/src/main/java/org/hisp/dhis/android/core/trackedentity/internal/TrackedEntityInstanceDownloadCall.kt

```
@@ -133,19 +138,68 @@ internal class TrackedEntityInstanceDownloadCall(
+        }
+
+        return when {
+            result is Result.Success -> result
+            result is Result.Failure &&
+                result.failure.errorCode() == D2ErrorCode.OWNERSHIP_ACCESS_DENIED -> result
+            result is Result.Failure &&
+                result.failure.errorCode() == D2ErrorCode.PROGRAM_ACCESS_CLOSED -> result
+            result is Result.Failure && result.failure.httpErrorCode() == HttpStatusCode.NotFound.value -> {
+                checkOwnershipOnNotFound(uid, query)
+            }
+            else -> result
+        }
+    }
+
+    private suspend fun checkOwnershipOnNotFound(
+        uid: String,
+        query: TrackerAPIQuery,
+    ): Result<TrackedEntityInstance?, D2Error> {
+        val program = query.commonParams.program ?: return Result.Success(null)
+
+        val queryWithoutProgram = TrackerAPIQuery(
+            commonParams = query.commonParams.copy(program = null),
+        )
+
+        val teiResult = coroutineCallExecutor.wrap(storeError = false) {
+            trackerCallFactory.getTrackedEntityCall().getEntityCall(uid, queryWithoutProgram)
+        }
+
+        return when (teiResult) {
+            is Result.Failure -> Result.Success(null)
+            is Result.Success -> {
+                val tei = teiResult.value
+                val programOwner = tei.programOwners()?.find { it.program() == program }
+
+                if (programOwner != null &&
+                    breakTheGlassHelper.isProtectedInSearchScope(program, programOwner.ownerOrgUnit())
+                ) {
+                    Result.Failure(
+                        D2Error.builder()
+                            .errorCode(D2ErrorCode.OWNERSHIP_ACCESS_DENIED)
+                            .errorDescription("OWNERSHIP_ACCESS_DENIED")
+                            .httpErrorCode(HttpStatusCode.NotFound.value)
+                            .build(),
+                    )
+                } else {
+                    Result.Success(null)
```

same here, I think it should return the original error?